api/v1beta1/openstackcontrolplane_types.go

```diff
@@ -22,35 +22,39 @@ import (
 
 // OpenStackControlPlaneSpec defines the desired state of OpenStackControlPlane
 type OpenStackControlPlaneSpec struct {
-	Controller OpenStackControllerSpec `json:"controller"`
+	// List of VirtualMachine roles
+	VirtualMachineRoles []OpenStackVirtualMachineRoleSpec `json:"virtualMachineRoles"`
```

Comment:
maybe we should use here a map instead of an array to be able to easy patch specific roles for kuttl tests when testing multiple VM roles

Reply:
done with latest commit